providers/src/airflow/providers/amazon/aws/hooks/sagemaker_unified_studio.py

```diff
@@ -0,0 +1,186 @@
+        config = ClientConfig()
+        config.overrides["execution"] = {"local": is_local_runner()}
+        return config
+
+    def _format_start_execution_input_config(self):
+        config = {
+            "notebook_config": {
+                "input_path": self.input_config.get("input_path"),
+                "input_parameters": self.input_config.get("input_params"),
+            },
+        }
+
+        return config
+
+    def _format_start_execution_output_config(self):
+        output_formats = (
+            self.output_config.get("output_formats") if self.output_config else ["NOTEBOOK"]
+        )
+        config = {
+            "notebook_config": {
+                "output_formats": output_formats,
+            }
+        }
+        return config
+
+    def start_notebook_execution(self):
+        start_execution_params = {
+            "execution_name": self.execution_name,
+            "execution_type": "NOTEBOOK",
+            "input_config": self._format_start_execution_input_config(),
+            "output_config": self._format_start_execution_output_config(),
+            "termination_condition": self.termination_condition,
+            "tags": self.tags,
+        }
+        if self.compute:
+            start_execution_params["compute"] = self.compute
+
+        request = StartExecutionRequest(**start_execution_params)
+
+        return self._sagemaker_studio.execution_client.start_execution(request)
+
+    def wait_for_execution_completion(self, execution_id, context):
+
+        while True:
+            time.sleep(self.poll_interval)
+            response = self.get_execution_response(execution_id)
+            error_message = response.get("error_details", {}).get("error_message")
+            status = response["status"]
+            if "files" in response:
+                self._set_xcom_files(response["files"], context)
+            if "s3_path" in response:
+                self._set_xcom_s3_path(response["s3_path"], context)
+
+            ret = self._handle_state(execution_id, status, error_message)
+            if ret:
+                return ret
+
+    def _set_xcom_files(self, files, context):
+        if not context:
+            error_message = "context is required"
+            raise AirflowException(error_message)
+        for file in files:
+            context["ti"].xcom_push(
+                key=f"{file['display_name']}.{file['file_format']}",
+                value=file["file_path"],
+            )
+
+    def _set_xcom_s3_path(self, s3_path, context):
+        if not context:
+            error_message = "context is required"
+            raise AirflowException(error_message)
+        context["ti"].xcom_push(
+            key="s3_path",
+            value=s3_path,
+        )
+
+    def get_execution_response(self, execution_id):
+        response = self._sagemaker_studio.execution_client.get_execution(
+            GetExecutionRequest(execution_id=execution_id)
+        )
+        return response
```

Comment:
```suggestion
        return self._sagemaker_studio.execution_client.get_execution(
            GetExecutionRequest(execution_id=execution_id)
        )
```

Any specific reason you're wrapping this one-liner in a method? I thought maybe for mocking purposes but I don't see anything like that below.